File: backend-ms1/src/main/java/ma/zs/stocky/service/facade/admin/commun/FournisseurAdminService.java
package ma.zs.stocky.service.facade.admin.commun;

import java.util.List;
import ma.zs.stocky.bean.core.commun.Fournisseur;
import ma.zs.stocky.dao.criteria.core.commun.FournisseurCriteria;
import ma.zs.stocky.zynerator.service.IService;

import ma.zs.stocky.ws.dto.commun.FournisseurDto;
import org.springframework.http.HttpEntity;


public interface FournisseurAdminService extends  IService<Fournisseur,FournisseurCriteria>  {



    HttpEntity<byte[]> createPdf(FournisseurDto dto) throws Exception;

}

File: frontend/src/app/controller/service/admin/commun/FournisseurAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import { RoleService } from 'src/app/zynerator/security/Role.service';
import {environment} from 'src/environments/environment';

import {FournisseurDto} from 'src/app/controller/model/commun/Fournisseur.model';
import {FournisseurCriteria} from 'src/app/controller/criteria/commun/FournisseurCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class FournisseurAdminService extends AbstractService<FournisseurDto, FournisseurCriteria> {
     constructor(private http: HttpClient, private roleService: RoleService) {
        super();
        this.setHttp(http);
        this.setApi(environment.apiUrl + 'admin/fournisseur/');
    }

    public constrcutDto(): FournisseurDto {
        return new FournisseurDto();
    }

    public constrcutCriteria(): FournisseurCriteria {
        return new FournisseurCriteria();
    }
}
